src/ydata_synthetic/synthesizers/gan.py

```diff
@@ -75,8 +88,34 @@ def model_name(self):
 
     def train(self,
               data: Union[DataFrame, array],
-              train_arguments: TrainParameters):
-        raise NotImplementedError
+              num_cols: Optional[List[str]] = None,
+              cat_cols: Optional[List[str]] = None,
+              preprocess: bool = True) -> Union[DataFrame, array]:
+        """Sets up the train session by instantiating an appropriate processor and transforming the data.
+        Returns the data object (preprocessed when applicable) for the child class specific train method to resume.
+        Args:
+            data (Union[DataFrame, array]): Raw data object.
+            num_cols (Optional[List[str]]): List of names of numerical columns.
+            cat_cols (Optional[List[str]]): List of names of categorical columns.
+            preprocess (bool): Determines if the preprocessor is to be run on the data or not (p.e. preprocessed data).
+        Returns:
+            processed_data (Union[DataFrame, array]): Processed data object (if applicable).
+        """
+        processed_data = data
+        if preprocess:
+            if self.__MODEL__ in RegularModels.__members__:
+                self.processor = RegularDataProcessor
+            elif self.__MODEL__ in TimeSeriesModels.__members__:
+                self.processor = TimeSeriesDataProcessor
+            else:
+                print(f'A DataProcessor is not available for the {self.__MODEL__}.')
+        if self.processor and preprocess:
+            processed_data = self.processor(num_cols, cat_cols).fit_transform(data)
+
+        self.data_dim = processed_data.shape[1]
```

Comment:
Move this logic to the synthesizers. As this logic can be different for time-series synths.